app/controllers/orders_controller.rb

```diff
@@ -5,15 +5,35 @@ def show
   end
 
   def create
-    order = current_user.orders.create
     if !@cart.contents.empty?
+      year, month, day = date_params
+      order = current_user.orders.new
       order.create_order(@cart)
-      session[:cart] = {}
-      flash[:alert] = "Order was successfully placed"
-      redirect_to "/orders"
+      order.assign_attributes(start_date: Date.new(year, month, day))
+      order.assign_attributes(end_date: order.start_date + 7.days)
+      order.cats.each do |cat|
+        if cat.orders.reserved?
+          flash[:notice] = "Sorry #{cat.name} is already reserved on that date."
+          redirect_to cart_path
+        else
+          order.save
+          session[:cart] = {}
+          flash[:alert] = "Order was successfully placed"
+          redirect_to "/orders"
+        end
```

Comment:
@adriennedomingus could you explain how `order.cats.each` works? 

I am not sure but it seems like if the first cat is not reserved we will save the order immediately. So if the last cat is reserved we will confirm the order.

Let me know what you think - Everything else looks good!